### PR TITLE
conditional processor execution

### LINKED ISSUE #1344: Add ability to specify condition when creating a processor

### LINKED ISSUE #1345: Add a field to the pipeline config file (bump minor version)

### PR DESCRIPTION
**Pull request by maha-hajja**
### Description

added a "condition" field for the processor instance, made sure to assign that field by gRPC and by pipeline config files when creating processors.
bumped the pipeline config file minor version from 2.0 to 2.1

Fixes #1344 
Fixes #1345 

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: pkg/orchestrator/orchestrator.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ type ConnectorService interface {
 type ProcessorService interface {
 	List(ctx context.Context) map[string]*processor.Instance
 	Get(ctx context.Context, id string) (*processor.Instance, error)
-	Create(ctx context.Context, id string, procType string, parent processor.Parent, cfg processor.Config, p processor.ProvisionType) (*processor.Instance, error)
+	Create(ctx context.Context, id string, procType string, parent processor.Parent, cfg processor.Config, p processor.ProvisionType, condition string) (*processor.Instance, error)
```
Comment (hariso) [inline pkg/orchestrator/orchestrator.go:110]: Nitpick: would be good to document what's expected to be in the string (a Go template)?

Alternatively, we could maybe have `type GoTemplate string`.

File: proto/api/v1/api.proto [hunk 2]
```diff
@@ -130,6 +130,7 @@ message Processor {
 
   string type = 4 [(google.api.field_behavior) = IMMUTABLE];
   Parent parent = 6 [(google.api.field_behavior) = IMMUTABLE];
+  string condition = 9;
```
Comment (hariso) [inline proto/api/v1/api.proto:133]: We should add some docs about the condition (i.e. that it's a Go template and whatever else is useful). I believe that adding the docs here would also put them into the Swagger docs.

**Review by hariso — COMMENTED** [review]
Looking good to me! Left a couple of nitpicks only!

File: pkg/orchestrator/orchestrator.go (hunk 1 above)
Comment (maha-hajja) [reply pkg/orchestrator/orchestrator.go:110]: i prefer adding docs, IMO adding more types will make it more complicated going forward with casting and type changing, I think "condition" is similar to the "type" field, which is also just a string

**Review by hariso — APPROVED** [review]

**Pull request merged**